src/script/interpreter.cpp

```diff
@@ -1757,6 +1757,17 @@ class CTransactionSignatureSerializer
     }
 };
 
+/** Compute the (single) SHA256 of the concatenation of all outpoint flags of a tx. */
+template <class T>
+uint256 GetOutpointFlagsSHA256(const T& txTo)
+{
+    CHashWriter ss(SER_GETHASH, 0);
+    for (const auto& txin : txTo.vin) {
+        ss << (unsigned char) ((!txin.assetIssuance.IsNull() << 7) + (txin.m_is_pegin << 6));
```

Comment:
Consider instead using `(COutPoint::OUTPOINT_ISSUANCE_FLAG >> 24)` `COutPoint::OUTPOINT_PEGIN_FLAG >> 24)`

I believe `|` is prefered over `+` for flag composition.